lib/streamlit/elements/heading.py

```diff
@@ -30,7 +33,7 @@
 class HeadingMixin:
     @gather_metrics("header")
     def header(
-        self, body: SupportsStr, anchor: Optional[str] = None
+        self, body: SupportsStr, anchor: Union[None, str, Literal[False]] = None
```

Comment:
Why don't we write `Optional[Union[str, Literal[False]]]`? It seems (a bit) more readable.

Reply:
It's a matter of code style, but in this case I've chosen to explicitly list all values of the valid value types, because each type has different behavior, so the type maps to that.
`None` - autogneerated anchor value,
`str` - used as-is
`Literal[False]` - hide anchor icon.